### PR TITLE
refactor: rework the hello hamlet content

### PR DESCRIPTION
**Pull request by roleyfoley**
## Description
Updates to the hello hamlet docs to follow a better flow, didn't actually add much new content just reviewed what was already there and tweaked it

## Target Audience
- [X] hamlet users
- [ ] hamlet framework developers (incl. plugins)

#### Documentation
- [ ] Spelling / Syntax correction only
- [X] Refactor (Documentation overhaul, or revising after changes to hamlet)
- [ ] New feature (Documenting something new)
#### Site Design
- [ ] Refactor (No new or removed content.)
- [ ] New Features & Content
- [ ] Docusaurus / Plugin version update.

## Followup Actions
- [x] None

## Checklist:
- [ ] I have tested the changes locally - see [README.md](https://github.com/hamlet-io/docs/blob/master/README.md)
- [ ] I have added appropriate labels to this PR.
- [ ] I have added this to the `hamlet roadmap` project.

### CODE REVIEW

File: docs/hello/hamlet.md [hunk 1]
```diff
@@ -5,37 +5,78 @@ title: Hello Hamlet
...
+Let's get started..
 
-## Building a Machine
+## An App
 
 <Mermaid chart={`
-    graph TD;
+    graph LR;
     container[Hello]
 `}/>
```
Comment (RossMurr4y) [inline docs/hello/hamlet.md:23]: Is there any way to set a size for this? It's showing tiny in the preview.

<image src="https://user-images.githubusercontent.com/20530524/101109359-d5e4cc80-362a-11eb-8a78-73a093f16213.png" height=300 />

File: docs/hello/hamlet.md [hunk 2]
```diff
@@ -5,37 +5,78 @@ title: Hello Hamlet
...
+
+Components have a **type** which defines the infrastructure required to provide this function. For hosting containers we have two types available, **service** which uses a container orchestrator (kubernetes, AWS ECS etc.) to ensure the container is always running, and **task** which run on demand, complete a specific task and then exit.
+We want to say `Hello!` all the time so we will make the type of this component a **service**.
+
+hamlet provides a library of components which are designed to work with the other components and can be used to deploy to multiple cloud providers. You can also provide your own component types and define their deployment to other providers as well. Each component type has it's own configuration that defines specifically how this component should behave, memory and cpu allocation, exposed network ports and connections to load balancers are all part of the component configuration
```
Comment (RossMurr4y) [inline docs/hello/hamlet.md:56]: Capitalise here?

**Comment by RossMurr4y** [comment]
Just a few minor niggles then this looks great!

I really like the tone you've set - typically I think the "guides" need to feel like this and have a more conversational and "fun" tone to them - I think you've hit a good balance.

I think we might need a very basic style guide sooner rather than later. You've begun **bolding** stuff that I would typically use `code` for - I don't care for either more than the other but consistency would be ideal. Thats another ticket though.

File: docs/hello/hamlet.md [hunk 3]
```diff
@@ -5,37 +5,78 @@ title: Hello Hamlet
...
 <Mermaid chart={`
-    graph TD;
+    graph LR;
     container[Hello]
 `}/>
 
-Here is our container, it performs a specific function within our application, saying `Hello!`. While it might have dependencies on other components, it still acts as an independent unit. In `hamlet` this unit is called a **component**, they form the basis of deploying infrastructure within `hamlet`.
-
-If we run this container locally and access its over http we meet our hello application and it lets us know where it is
+Here is our widget, it performs a specific function, saying `Hello!`. While it might be part of a bigger application, it still acts as an independent unit. If we run this container locally and access it over http we meet our hello widget and it lets us know where it is
```
[the remark this answers is not in the document]
Comment (ken9ross) [reply docs/hello/hamlet.md:25]: I am deliberately removing the term widget, as I don't think it's needed (we struggle to find sufficient terms for our needs, and we're throwing a large number of new terms at the reader)

File: docs/hello/hamlet.md [hunk 4]
```diff
@@ -5,37 +5,78 @@ title: Hello Hamlet
...
+curl http://localhost:8000
+{
+  "Greeting": "Hello!",
+  "Location": "world"
+}
+```
+
+## A Component
+
+So we've seen our widget and what it does when we run it locally, now we want to to make this widget available as pat of our application. In hamlet this widget, a specific function performed in your application is called a **component**, they form the basis of deploying infrastructure within hamlet.
```
Comment (ken9ross) [inline docs/hello/hamlet.md:51]:
```suggestion
So now we want to to make this functionality available as pat of our application. In hamlet **component**s form the basis of deploying infrastructure.
```

File: docs/hello/hamlet.md [hunk 5]
```diff
@@ -5,37 +5,78 @@ title: Hello Hamlet
...
+
+Components have a **type** which defines the infrastructure required to provide this function. For hosting containers we have two types available, **service** which uses a container orchestrator (kubernetes, AWS ECS etc.) to ensure the container is always running, and **task** which run on demand, complete a specific task and then exit.
```
Comment (ken9ross) [inline docs/hello/hamlet.md:53]:
```suggestion
Components have a **type** which defines the infrastructure required to provide this function. For hosting containers we have two types available:
- **service** which uses a container orchestrator (kubernetes, AWS ECS etc.) to ensure the container is always running, and
- **task** which run on demand, complete a specific task and then exit.

```

File: docs/hello/hamlet.md (hunk 2 above)
Comment (ken9ross) [inline docs/hello/hamlet.md:56]:
```suggestion
Hamlet provides a library of standard components which are designed to work with other components and deploy to multiple cloud providers. You can also provide your own component types and define their deployment to other providers as well. Each component type has its own configuration that defines how this component should behave. Memory and cpu allocation, exposed network ports and connections to load balancers are all parts of the component configuration
```

File: docs/hello/hamlet.md [hunk 6]
```diff
@@ -5,37 +5,78 @@ title: Hello Hamlet
...
+                        "MemoryReservation" : "64",
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+If we wanted to say `Hello!` from multiple locations we would need to deploy multiple widgets and update the location for each one. Our component is performing the same function with different run time configuration to provide different features.
```
Comment (ken9ross) [inline docs/hello/hamlet.md:78]:
```suggestion
## Component Reuse

If we wanted to say `Hello!` from multiple locations we would need to deploy multiple widgets and update the location for each one. Our component is performing the same function with different run time configuration to provide different features. This is obviously inefficient.
```

File: docs/hello/hamlet.md [hunk 7]
```diff
@@ -55,68 +96,12 @@ A given instance and version of a component is called an **occurrence**.
...
-        "NetworkMode" : "<string>"
-    }
-}
-```
 
-In most IT systems we don't just deploy one component, we deploy a collection of components which work together to provide a specific service. In `hamlet` this collection is called a **Solution**, the components that make up the solution have a set of dependent relationships between one another and we define these relationships between as **Links**. Links allow components to share properties between one another and can also be used to determine the configuration of a component.
+## A link
 
-To make sure we can say `Hello!` all of the time we configure our service to deploy 2 copies of the same container, and so we can access both of them using the same URL we add a load balancer to our **Solution**
+In an application, we don't just deploy one component, we deploy a collection of components which work together to provide a specific purpose, in hamlet this collection is called a **Solution**.
```
Comment (ken9ross) [inline docs/hello/hamlet.md:104]:
```suggestion
In an application, we don't just deploy one component, we deploy a collection of components which work together to provide a specific purpose, in hamlet this collection is called a **Solution**. Within a Solution, the components will have relationships to other components.
```

File: docs/hello/hamlet.md [hunk 8]
```diff
@@ -129,6 +114,4 @@ To make sure we can say `Hello!` all of the time we configure our service to dep
     end
 `}/>
 
-
-
-To create this relationship in `hamlet` we create a link from the `Hello` component to the `Hello-LB` component. When we deploy this infrastructure the container will be configured to register with the load balancer which will send traffic to either copy of the `Hello` Service.
+To create this relationship in hamlet we create a link from the `Hello` component to the `Hello-LB` component. When we deploy this infrastructure the container will be configured to register with the load balancer which will send traffic to either copy of the `Hello` Service.
```
Comment (ken9ross) [inline docs/hello/hamlet.md:117]:
```suggestion
To create this relationship in hamlet we create a link between components. In this case, from the `Hello` component to the `Hello-LB` component. When we deploy this infrastructure the container will be configured to register with the load balancer which will send traffic to either copy of the `Hello` Service.
```

**Review by ken9ross — COMMENTED** [review]
Not sure if this is what you wanted or not, but my $0.02 worth is attached

File: docs/hello/hamlet.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (roleyfoley) [reply docs/hello/hamlet.md:25]: I used widget as its a common term used in a broader sense. Piece of functionality feels really long when a widget is pretty much that anway

**Pull request merged**